search/task_helpers.py

```diff
@@ -0,0 +1,139 @@
+
+    Args:
+        post_obj (praw.models.reddit.submission.Submission): A PRAW post ('submission') object
+    """
+    data = {
+        'object_type': POST_TYPE,
+        'author': post_obj.author.name,
+        'channel_title': post_obj.subreddit.display_name,
+        'text': post_obj.selftext,
+        'score': post_obj.score,
+        'created': post_obj.created,
+        'post_id': post_obj.id,
+        'post_title': post_obj.title,
+        'num_comments': post_obj.num_comments,
+    }
+    return create_document.delay(
+        gen_post_id(post_obj.id),
+        data
+    )
+
+
+def update_post_text(post_obj):
+    """
+    Serializes post object text and runs a task to update the text for the associated ES document.
+
+    Args:
+        post_obj (praw.models.reddit.submission.Submission): A PRAW post ('submission') object
+    """
+    return update_document_with_partial.delay(
+        gen_post_id(post_obj.id),
+        {'text': post_obj.selftext}
+    )
+
+
+def update_post_removal_status(post_obj):
+    """
+    Serializes the removal status for a post object and runs a task to update that status
+    for the associated ES document.
+
+    Args:
+        post_obj (praw.models.reddit.submission.Submission): A PRAW post ('submission') object
+    """
+    return update_document_with_partial.delay(
+        gen_post_id(post_obj.id),
+        {'removed': bool(post_obj.banned_by) and not post_obj.approved_by}
+    )
+
+
+def _update_post_comment_count(comment_obj, incr_amount=1):
+    """
+    Updates the comment count for a post object (retrieved via a comment object)
+    and runs a task to update that count for the associated ES document.
+
+    Args:
+        comment_obj (praw.models.reddit.comment.Comment): A PRAW comment object
+        incr_amount (int): The amount to increment to count
+    """
+    post_obj = comment_obj.submission
+    return increment_document_integer_field.delay(
+        gen_post_id(post_obj.id),
+        field_name='num_comments',
+        incr_amount=incr_amount
+    )
+
+
+increment_post_comment_count = partial(_update_post_comment_count, incr_amount=1)
+decrement_post_comment_count = partial(_update_post_comment_count, incr_amount=-1)
+
+
+def update_indexed_score(instance, instance_type, vote_action=None):
+    """
+    Serializes the score for a PRAW object (comment or post) and runs a task to update
+    that score for the associated ES document.
+
+    Args:
+        instance: A PRAW Comment or Submission (a.k.a. post) object
+        instance_type (str): A string indicating the type of object being passed in
+        vote_action (str): A string indicating what kind of vote action was taken
+    """
+    if vote_action == VOTE_ACTIONS.UPVOTE or vote_action == VOTE_ACTIONS.CLEAR_DOWNVOTE:
+        vote_increment = 1
+    elif vote_action == VOTE_ACTIONS.DOWNVOTE or vote_action == VOTE_ACTIONS.CLEAR_UPVOTE:
+        vote_increment = -1
+    else:
+        return
```

Comment:
Should this raise an exception instead of quietly returning? Passing in an unsupported value here should be programmer error and we'd want to catch that.